### DISCUSSION TITLE
Alias `:name`/`:csv` with `:keys`/`:values`

### DISCUSSION THREAD

**Discussion by Pyrolistical** [vitaly-t/pg-promise#892] [Ideas]
consider this insert
```js
await tx.none(
  `
    INSERT INTO "table"(
      $/row:name/
    )
    VALUES(
      $/row:csv/
    )
  `,
  {
    row: {
      column_1: 'value 1',
      column_2: 'value 2'
    }
  }
);
```

I think it would be more obvious we if we could write
```js
await tx.none(
  `
    INSERT INTO "table"(
      $/row:keys/
    )
    VALUES(
      $/row:values/
    )
  `,
  {
    row: {
      column_1: 'value 1',
      column_2: 'value 2'
    }
  }
);
```

**Comment by vitaly-t** [comment]
You are formatting SQL Names, not SQL keys, i.e. you are confusing source with destination, i.e. each filter is named according to the expected result, not the type of source.

**Reply by Pyrolistical** [reply]
With pure functions, as these filters are, naming by either expected input or output is equivalent. Just purely an api design/stylistic choice. 

IMO adding an alias is more inclusive to people who start with the input or start with the desired result first

**Reply by vitaly-t** [reply]
For stylistics, you can just wrap the query into a reusable function:

```ts
function insertObj(table: string, obj: any) {
    return pgp.as.format('INSERT INTO $1:name($2:name) VALUES($2:csv)', [table, obj]);
}

// and then use it like so:

await tx.none(insertObj('table', row));
```

**Comment by vitaly-t** [comment]
Also, a little simpler syntax:

```ts
await tx.none(
  `
    INSERT INTO "table"(
      $/this:name/
    )
    VALUES(
      $/this:csv/
    )
  `,
  {
      column_1: 'value 1',
      column_2: 'value 2'
    }
);
````
